rest/route/github.go

```diff
@@ -763,6 +767,50 @@ func (gh *githubHookApi) createVersionForTag(ctx context.Context, pRef model.Pro
 			"tag":                tag,
 			"message":            "user not authorized for git tag version",
 		})
+		metadata := model.VersionMetadata{
+			Revision: revision,
+			GitTag:   tag,
+		}
+		stubVersion, dbErr := repotracker.ShellVersionFromRevision(ctx, &pRef, metadata)
+		if dbErr != nil {
+			grip.Error(message.WrapError(dbErr, message.Fields{
+				"message":            "error creating shell version",
+				"project":            pRef.Id,
+				"project_identifier": pRef.Identifier,
+				"revision":           revision,
+			}))
+		}
+		stubVersion.Errors = []string{errors.Errorf("user '%s' not authorized for git tag version", tag.Pusher).Error()}
+		err := stubVersion.Insert()
+		if err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":            "error inserting stub version for failed git tag version",
+				"project":            pRef.Id,
+				"project_identifier": pRef.Identifier,
+				"revision":           revision,
+			}))
+		}
+		event.LogVersionStateChangeEvent(stubVersion.Id, evergreen.VersionFailed)
+		userDoc, err := user.FindByGithubName(tag.Pusher)
+		if err != nil {
+			return nil, errors.Wrapf(err, "finding user '%s'", tag.Pusher)
+		}
+		if userDoc != nil {
+			sender, err := evergreen.GetEnvironment().GetSender(evergreen.SenderEmail)
```

Comment:
Should we make this configurable here: https://spruce.mongodb.com/preferences/notifications? 
I worry that emails are more likely to get lost than slack messages.

Reply:
Imo since we're also creating an errored stub version that will show up on the project health page, the extra steps to modify our resolvers + UI for this can be forgone for the time being, and revisited if there continues to be a discoverability issue.